molecule-runtime/src/commonMain/kotlin/app/cash/molecule/molecule.kt

```diff
@@ -133,17 +151,43 @@ public fun <T> CoroutineScope.launchMolecule(
   mode: RecompositionMode,
   emitter: (value: T) -> Unit,
   body: @Composable () -> T,
+) {
+  launchMolecule(
+    context = coroutineContext,
+    mode = mode,
+    emitter = emitter,
+    body = body,
+  )
+}
+
+/**
+ * Launch a coroutine into this [CoroutineScope] which will continually recompose `body`
+ * in the optional [context] to invoke [emitter] with each returned [T] value.
+ *
+ * [launchMolecule]'s [emitter] is always free-running and will not respect backpressure.
+ * Use [moleculeFlow] to create a backpressure-capable flow.
+ *
+ * Accepts a [context] to launch under.
+ */
+public fun <T> CoroutineScope.launchMolecule(
+  context: CoroutineContext,
+  mode: RecompositionMode,
+  emitter: (value: T) -> Unit,
+  body: @Composable () -> T,
 ) {
   val clockContext = when (mode) {
     RecompositionMode.ContextClock -> EmptyCoroutineContext
     RecompositionMode.Immediate -> GatedFrameClock(this)
   }
 
   with(this + clockContext) {
-    val recomposer = Recomposer(coroutineContext)
+    val recomposer = Recomposer(context)
```

Comment:
Yes, the scope context needs combined with the external context which needs amended with the clock context. Clock should be last since we want the immediate clock to take precedence over any clock registered in the external context.

Reply:
If you don't get to it, I'll take over the PR tomorrow and make the necessary changes (along with some style nits) and land it. Mostly out today.

Reply:
Thanks, [updated!](https://github.com/cashapp/molecule/pull/312/commits/e9b89457226d881c5403ac4921f059994cf77026)!